sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/MoveVariableInsideIfCheck.java

```diff
@@ -0,0 +1,275 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.ScopeUtils;
+
+/**
+ * <p>
+ * Checks if a variable is only used inside if statements and asks for it's
+ * declaration to be moved there too.
+ * </p>
+ *
+ * @author Richard Veach
+ */
+public class MoveVariableInsideIfCheck extends AbstractCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "move.variable.inside";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {TokenTypes.VARIABLE_DEF};
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ScopeUtils.isLocalVariableDef(ast)) {
+            validateLocalVariable(ast);
+        }
+    }
+
+    /**
+     * Examines the local variable for violations to be moved inside an nest if
+     * statement.
+     *
+     * @param ast The local variable to examine.
+     */
+    private void validateLocalVariable(DetailAST ast) {
+        final Holder holder = new Holder(ast);
+
+        for (DetailAST child = ast.getNextSibling(); (!holder.exit) && (child != null);
+                child = child.getNextSibling()) {
+            switch (child.getType()) {
+                case TokenTypes.LITERAL_IF:
+                    validateIf(holder, child);
+                    break;
+                default:
+                    validateOther(holder, child);
+                    break;
+            }
+        }
+
+        if (holder.blockNode != null) {
+            log(ast, MSG_KEY, holder.variableName, holder.blockNode.getLineNo());
+        }
+    }
+
+    /**
+     * Examines an if statement to see how many times the specified variable
+     * identifier was used inside it.
+     *
+     * @param holder The object holder with the specified variable to check and
+     *        it's current state.
+     * @param ifNodeGiven The current if node to examine.
+     */
+    private static void validateIf(Holder holder, DetailAST ifNodeGiven) {
+        DetailAST ifNode = ifNodeGiven;
+
+        // -@cs[SingleBreakOrContinue] Too complex to break apart
+        while (true) {
```

Comment:
loop is going through all `ifs` of `if-else` chain.
while loop could be changed to make sure if node is an if node.

However, the suppression is for the number of breaks and this change won't help that.
I think it will hurt code readability by changing it, but thats just my opinion.